src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        if (fromFileName == null || fromFileName.isEmpty()
+                || toFileName == null || toFileName.isEmpty()) {
+            throw new RuntimeException("Invalid file name");
+        }
+        String fileContent = fileToString(fromFileName);
+        String[] separatedItems = fileContent.split(System.lineSeparator());
+        int buy = 0;
+        int supply = 0;
+        for (String item : separatedItems) {
+            String[] splited = item.split(",");
+            if (splited[0].equals("buy")) {
+                buy += Integer.parseInt(splited[1]);
+            } else {
+                supply += Integer.parseInt(splited[1]);
+            }
+        }
+        final int result = supply - buy;
+        createReport(buy, supply, result, toFileName);
```

Comment:
```suggestion
        String[] separatedItems = fileContent.split(System.lineSeparator());
        int buy = 0;
        int supply = 0;
        for (String item : separatedItems) {
            String[] splited = item.split(",");
            if (splited[0].equals("buy")) {
                buy += Integer.parseInt(splited[1]);
            } else {
                supply += Integer.parseInt(splited[1]);
            }
        }
        final int result = supply - buy;
        createReport(buy, supply, result, toFileName);
```

Move the report creation logic into a separate method